sky/resources.py

```diff
@@ -55,12 +55,15 @@ def __init__(
         spot_recovery: Optional[str] = None,
         disk_size: Optional[int] = None,
         region: Optional[str] = None,
+        zone: Optional[str] = None,
         image_id: Optional[str] = None,
     ):
         self._version = self._VERSION
         self._cloud = cloud
         self._region: Optional[str] = None
+        self._zone: Optional[str] = None
         self._set_region(region)
+        self._set_zone(zone)
```

Comment:
We can use a single `self.validate_region_zone` function below this line, if we combined the `region_exists` and `zone_exists` functions.

Reply:
Added. see comments above.